### PR TITLE
Add mir building test directory

### PR DESCRIPTION
**Pull request by JakobDegen**
The first commit renames `mir-map.0` mir dumps to `built.after` dumps. I am happy to drop this commit if someone can explain the origin of the name.

The second commit moves a bunch of mir building tests into their own directory. I did my best to make sure that all of these tests are actually testing mir building, and not just incidentally using `built.after`

r? @oli-obk

### CODE REVIEW

**Comment by oli-obk** [comment]
I thought we were already dumping right after mir building (although with a weird name)

**Comment by JakobDegen** [comment]
Yeah, `PromoteTemps.before`. I can just switch to using that if you want, but I figured some clear naming might be nice

**Comment by cjgillot** [comment]
There is the `mir_map` dump.  Should it be removed?

**Comment by oli-obk** [comment]
> There is the `mir_map` dump. Should it be removed?

This is the one I meant.

**Comment by JakobDegen** [comment]
Gonna wait on #102340 before fixing this

**Comment by JakobDegen** [comment]
Rebased, made changes, and updated top comment of PR.

@rustbot ready

File: src/test/mir-opt/spanview_block.main.built.after.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<!DOCTYPE html>
```
Comment (oli-obk) [inline src/test/mir-opt/spanview_block.main.built.after.html:1]: I think you need to delete the old HTML files

**Pull request merged**